crypto/asn1/tasn_enc.c

```diff
@@ -79,20 +80,20 @@ static int asn1_item_flags_i2d(ASN1_VALUE *val, unsigned char **out,
  * performs the normal item handling: it can be used in external types.
  */
 
-int ASN1_item_ex_i2d(ASN1_VALUE **pval, unsigned char **out,
+int ASN1_item_ex_i2d(const ASN1_VALUE **pval, unsigned char **out,
                      const ASN1_ITEM *it, int tag, int aclass)
 {
     const ASN1_TEMPLATE *tt = NULL;
     int i, seqcontlen, seqlen, ndef = 1;
     const ASN1_EXTERN_FUNCS *ef;
     const ASN1_AUX *aux = it->funcs;
-    ASN1_aux_cb *asn1_cb = 0;
+    ASN1_aux_const_cb *asn1_cb = 0;
 
     if ((it->itype != ASN1_ITYPE_PRIMITIVE) && !*pval)
         return 0;
 
     if (aux && aux->asn1_cb)
```

Comment:
Should this be `aux->asn1_const_cb` here?

Reply:
`asn1_cb` is just a local variable here. I left its name as it was for now just to minimize clutter for the review. Yet of course we can rename it to `asn1_const_cb`, or maybe simply `cb`.

Reply:
@mattcaswell, now I realize that I misunderstood your comment, which was not about the variable name but about which field to address.
Yes, there was a mistake, which I (re-)discovered independently and corrected in conjunction with my most recent changes (introducing backward compatibility as suggested by @levitte).